src/views/shared/amount-input/amount-input.view.jsx

```diff
@@ -101,6 +99,7 @@ function AmountInput (Component) {
      * @param {InputEvent} event - Input event
      */
     function handleInputChange (event) {
+      const INPUT_REGEX = new RegExp(`^\\d*(?:\\.\\d{0,${account?.token.decimals}})?$`)
```

Comment:
Why moving this regex to the input change? It's better to have it as a constant at the top as it was before to prevent creating it at every input change. Also the `?` is not needed as when we create the account we are going to add the token.

Reply:
The safe call "?" cover a bug with token swap because render it without know an account, and we moved it inside because was a bug in the matic swapper that was copied from this example

Reply:
Which was the bug of having the regex outside this function?

Reply:
The rule needs know the maximum decimals of the token to not break the bignumber format